### PR TITLE
Sovereign accounts parser

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Sovereign accounts parser that creates initial txs in sovereign shard
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by raduchis — CHANGES_REQUESTED** [review]

File: genesis/parsing/errors.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package parsing
+
+import "errors"
+
+var errNilAccountsParser = errors.New("nil accounts parser provided")
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply genesis/parsing/errors.go]: Right, removed this one and used suggested one

**Review by sasurobert — APPROVED** [review]

File: genesis/parsing/sovereignAccountsParser.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+func (ap *sovereignAccountsParser) GenerateInitialTransactions(
+	shardCoordinator sharding.Coordinator,
+	indexingData map[uint32]*genesis.IndexingData,
+) ([]*block.MiniBlock, map[uint32]*outportcore.TransactionPool, error) {
+	if check.IfNil(shardCoordinator) {
+		return nil, nil, genesis.ErrNilShardCoordinator
+	}
+
+	shardIDs := make([]uint32, 1)
+	shardIDs[0] = core.SovereignChainShardId
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply genesis/parsing/sovereignAccountsParser.go]: Yes, it is the same, changed as suggested

**Review by raduchis — APPROVED** [review]

File: genesis/parsing/errors.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply genesis/parsing/errors.go]: ok

File: genesis/parsing/sovereignAccountsParser.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply genesis/parsing/sovereignAccountsParser.go]: ok

**Pull request merged**